x-pack/plugin/ml/qa/semantic-search-tests/src/yamlRestTest/java/org/elasticsearch/smoketest/SemanticSearchNoReadPermissionsIT.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+package org.elasticsearch.smoketest;
+
+import com.carrotsearch.randomizedtesting.annotations.Name;
+
+import org.elasticsearch.test.rest.yaml.ClientYamlTestCandidate;
+import org.elasticsearch.test.rest.yaml.section.DoSection;
+import org.elasticsearch.test.rest.yaml.section.ExecutableSection;
+
+import java.io.IOException;
+
+import static org.hamcrest.Matchers.containsString;
+
+public class SemanticSearchNoReadPermissionsIT extends AbstractSemanticSearchPermissionsIT {
+
+    private static final String USERNAME = "no_read_index_no_ml";
+
+    private final ClientYamlTestCandidate testCandidate;
+
+    public SemanticSearchNoReadPermissionsIT(@Name("yaml") ClientYamlTestCandidate testCandidate) {
```

Comment:
This is for my education: I guess `@Name("yaml")` is what ensures only the newly added `10_basic.yml` file gets to run through the test suite? i.e. no other existing yaml tests will be executed by this class?

Reply:
By default all the yaml tests under the `resources/rest-api-spec/` are run, if you want to pull in other yaml tests that is configured in the build.gradle file. Here's an example

https://github.com/elastic/elasticsearch/blob/3a9e511117b561d51e7c5ade7508e15f718d1387/qa/ccs-common-rest/build.gradle#L33
 

I believe the `@Name` used to add extra details when reporting a test failure.